library/data/recipe/src/main/java/org/quiltmc/qsl/recipe/api/RecipeLoadingEvents.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.recipe.api;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.recipe.Recipe;
+import net.minecraft.recipe.RecipeType;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.base.api.event.EventAwareListener;
+
+/**
+ * Represents the recipe loading events.
+ * <p>
+ * Triggered when the recipes are loaded.
+ *
+ * @version 1.0.0
+ * @since 1.0.0
+ */
+public final class RecipeLoadingEvents {
+	/**
+	 * Recipe loading event, triggered when the recipes are loaded.
+	 */
+	public static final Event<RecipeLoadingCallback> REGISTER = Event.create(RecipeLoadingCallback.class,
+			callbacks -> handler -> {
+				for (var callback : callbacks) {
+					callback.onLoadRecipe(handler);
+				}
+			});
+	public static final Event<RecipeModifyCallback> MODIFY = Event.create(RecipeModifyCallback.class,
+			callbacks -> handler -> {
+				for (var callback : callbacks) {
+					callback.onModifyRecipe(handler);
+				}
+			});
+	public static final Event<RecipeRemoveCallback> REMOVE = Event.create(RecipeRemoveCallback.class,
+			callbacks -> handler -> {
+				for (var callback : callbacks) {
+					callback.onRecipeRemovePhase(handler);
+				}
+			});
+
+	private RecipeLoadingEvents() {
+		throw new UnsupportedOperationException("RecipeLoadingEvents only contains static definitions.");
+	}
+
+	/**
+	 * Callback called to register additional recipes when recipes are loaded.
+	 */
+	@FunctionalInterface
+	public interface RecipeLoadingCallback extends EventAwareListener {
+		/**
+		 * Called when recipes are loaded.
+		 * <p>
+		 * {@code handler} is used to add recipes into the {@linkplain net.minecraft.recipe.RecipeManager recipe manager}.
+		 *
+		 * @param handler the recipe handler
+		 */
+		void onLoadRecipe(RecipeHandler handler);
```

Comment:
The javadoc implies that this is ran once when all recipes are loaded, but the method name implies this is called each time a recipe is loaded. Please clarify both.